src/body-parts/BodyPart.cpp

```diff
@@ -0,0 +1,233 @@
+// Constructors
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/**
+ * Constructor.
+ *
+ * @param position The position of the body part
+ */
+BodyPart::BodyPart(const Vector4& position)
+{
+    _position = position;
+
+    _startLinesVerticesBufferStartIndex = BufferManager::add(LINES_VERTICES, _linesVertices);
+    _startLinesColorBufferStartIndex = BufferManager::add(LINES_COLORS, _linesColors);
+    _startTrianglesVerticesBufferStartIndex = BufferManager::add(TRIANGLES_VERTICES, _trianglesVertices);
+    _startTrianglesColorBufferStartIndex = BufferManager::add(TRIANGLES_COLORS, _trianglesColors);
+    _updateVertices(false);
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Getters
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/**
+ * @return The position of the body part
+ */
+[[nodiscard]] Vector4 BodyPart::getPosition() const
+{
+    return _position;
+}
+
+/**
+ * @return The direction of the body part
+ */
+[[nodiscard]] Vector4 BodyPart::getDir() const
+{
+    return _dir;
+}
+
+/**
+ * @return The children of the body part
+ */
+[[nodiscard]] std::vector<BodyPart*> BodyPart::getChildren() const
+{
+    return _children;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Setters
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/**
+ * Set the position of the body part.
+ *
+ * @param position The new position of the body part.
+ */
+void BodyPart::setPosition(const Vector4& position)
+{
+    _position = position;
+    _updateVertices();
+}
+
+/**
+ * Set the direction of the body part.
+ *
+ * @param dir The new position of the body part.
+ */
+void BodyPart::setDir(const Vector4& dir)
+{
+    constexpr float pi = std::numbers::pi;
+    _dir.setX(fmodf(dir.getX() + pi, 2 * pi) - pi);
+    _dir.setY(fmodf(dir.getY() + pi, 2 * pi) - pi);
+    _dir.setZ(fmodf(dir.getZ() + pi, 2 * pi) - pi);
+    _updateVertices();
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Methods
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/**
+ * Add a child to the body part.
+ *
+ * @param child The child to add
+ */
+void BodyPart::addChild(BodyPart* child)
+{
+    _children.push_back(child);
+}
+
+/**
+ * Remove a child from the body part.
+ *
+ * @param child The child to remove
+ */
+void BodyPart::removeChild(const BodyPart* child)
+{
+    std::erase(_children, child);
+}
+
+/**
+ * Update the vertices of the body part.
+ *
+ * @param modifyBuffer If true, the buffer will be modified
```

Comment:
Dans `Vector4.cpp` tu as mis `* @param w The w component of the vector (default to 1.0f)`, tu peux rajouter le `(default to true)` ici aussi stp?